controller/workitem.go

```diff
@@ -259,6 +259,30 @@ func ConvertJSONAPIToWorkItem(ctx context.Context, method string, appl applicati
 			target.Fields[workitem.SystemAssignees] = ids
 		}
 	}
+	if source.Relationships != nil && source.Relationships.Labels != nil {
+		if source.Relationships.Labels.Data == nil {
+			delete(target.Fields, workitem.SystemLabels)
+		} else {
+			distinctIDs := make(map[string]struct{})
+			for _, d := range source.Relationships.Labels.Data {
+				labelUUID, err := uuid.FromString(*d.ID)
+				if err != nil {
+					return errors.NewBadParameterError("data.relationships.assignees.data.id", *d.ID)
```

Comment:
A test for this would be good.

Reply:
Added a test in here https://github.com/fabric8-services/fabric8-wit/pull/1627/commits/b00281c799bf5144966a79fb2095631c8aaa122a